### PR TITLE
Be able to search for a port and have the relevant service shown

### PR DESCRIPTION
**Pull request by Smudgey**
For issue #1

### CODE REVIEW

File: app/services/SMService.scala [hunk 1]
```diff
@@ -93,14 +92,23 @@ trait SMService extends Logging {
...
+      case (_, true) => exactMatchSearch(servicesJson, query.toUpperCase)
+      case _ => servicesJson.collect { case (name, _) if name.contains(query.toUpperCase) => name }
     }
   }
 
+  private def isAnInt(query: String): Boolean = Try(query.toInt).isSuccess
+
+  private def searchServicesUsingPort(query: String, servicesJson: => Seq[(String, JsValue)]): Seq[String] =
+    servicesJson
+      .filter { case (_, js) => js.\("defaultPort").asOpt[Int].isDefined }
```
Comment (ChrisHillCap) [inline app/services/SMService.scala]: Does this line do anything? I think the collect covers this :)

File: app/services/SMService.scala [hunk 2]
```diff
@@ -93,14 +92,23 @@ trait SMService extends Logging {
...
-      exactMatchSearch(jsonConnector.loadServicesJson.fields, query)
-    } else {
-      jsonConnector.loadServicesJson.fields.collect { case (name, _) if name.contains(query) => name}
+  def searchForService(query: String): Seq[String] = {
+    val servicesJson: Seq[(String, JsValue)] = jsonConnector.loadServicesJson.fields
+
+    (isAnInt(query), query.contains("\"")) match {
+      case (true, _) => searchServicesUsingPort(query, servicesJson)
+      case (_, true) => exactMatchSearch(servicesJson, query.toUpperCase)
+      case _ => servicesJson.collect { case (name, _) if name.contains(query.toUpperCase) => name }
```
Comment (ChrisHillCap) [inline app/services/SMService.scala:100]: servicesJson.filter(_._1.contains(query.toUpperCase)) could work aswell

Comment (Smudgey) [reply app/services/SMService.scala:100]: Yep, that would work and would return more data than we are currently looking for (Seq[String] vs Seq[(String, JsValue)]). However, in the context of how searchForService is used, we only need the service name. The only change to that line from before was to transform the query to uppercase.

File: app/services/SMService.scala (hunk 1 above)
Comment (Smudgey) [reply app/services/SMService.scala]: Removed this line 👍

File: app/services/SMService.scala [hunk 3]
```diff
@@ -93,14 +92,21 @@ trait SMService extends Logging {
...
     }
   }
 
+  private def isAnInt(query: String): Boolean = Try(query.toInt).isSuccess
+
+  private def searchServicesUsingPort(query: String, servicesJson: => Seq[(String, JsValue)]): Seq[String] =
+    servicesJson.collect { case (name, js) if js.\("defaultPort").as[Int].toString.contains(query) => name }
```
Comment (chrisjwwalker) [inline app/services/SMService.scala]: Where you get the default port and use `as[Int]`, i've found it is failing for me as it's trying to process `MONGO` as a service. Unfortunately mongo doesn't have a port defined. Switching the `as[Int]` to `asOpt[Int]` should work

**Pull request merged**